lib/rack/attack/store_proxy/redis_cache_store_proxy.rb

```diff
@@ -15,17 +15,33 @@ def increment(name, amount = 1, options = {})
           #
           # So in order to workaround this we use RedisCacheStore#write (which sets expiration) to initialize
           # the counter. After that we continue using the original RedisCacheStore#increment.
-          if options[:expires_in] && !read(name)
-            write(name, amount, options)
+          rescuing do
+            if options[:expires_in] && !read(name)
+              write(name, amount, options)
 
-            amount
-          else
-            super
+              amount
+            else
+              super
+            end
           end
         end
 
+        def read(name)
+          rescuing { super(name) }
```

Comment:
I think we are overriding [`ActiveSupport::Cache::Store#read(name, options = nil)`](https://github.com/rails/rails/blob/b9ca94caea2ca6a6cc09abaffaad67b447134079/activesupport/lib/active_support/cache.rb#L340), which potentially can receive two arguments.

What about going with the following?

```ruby
def read(*_args)
  rescuing { super }
end
```

So that we allow `options` argument, plus we make it forward compatible with any arity changes to the original `#read`?

Reply:
That looks sensible, I've updated it!